Block/Entity/Item Mod Support for some mods for Unstable Branch

Excederus opened the pull request:
Added support for the following mods:
- Bountiful Blocks (block)
- Corpse (entity)
- Connected Glass (block)
- Countered's Terrain Slabs (block)
- Creeper Overhaul (block, entity)
- Dynamic Trees Plus (block)
- Fast Item Frames (block)
- Fast Paintings (block)
- Fowl Play (entity)
- Its Only Natural (block, item)
- Luminous Beasts (block, entity, item)
- Luminous Monsters (entity)
- MultiBeds (block)
- Project: Vibrant Journeys (block, item)
- RealmRPG: Fallen Adventurers (block)
- Resource Backpack's (block)
- Serene Seasons (block)
- Sooty Chimneys (block)

Excederus committed:
Block/Entity/Item Mod Support for some mods

Added support for the following mods:
- Bountiful Blocks (block)
- Corpse (entity)
- Countered's Terrain Slabs (block)
- Creeper Overhaul (block, entity)
- Dynamic Trees Plus (block)
- Fast Item Frames (block)
- Fast Paintings (block)
- Fowl Play (entity)
- Its Only Natural (block, item)
- Luminous Beasts (block, entity, item)
- Luminous Monsters (entity)
- MultiBeds (block)
- Project: Vibrant Journeys (block, item)
- RealmRPG: Fallen Adventurers (block)
- Resource Backpack's (block)
- Serene Seasons (block)
- Sooty Chimneys (block)

Excederus committed:
Merge branch 'X0nk:Unstable' into Unstable

Excederus commented:
I synced my fork after pushing the changes, so it's 2 commits, but shouldn't be an issue I hope

MikiP98 commented:
Two small comments from me:
1. From a quick look it seems the mod alphabetical ordering is broken. A.K.A. mods for each block.x entry should be ordered alphabetically one under another. It’s a limitation introduced by me with no official comment from Xonx, though I think it is easy to follow convention that makes looking at and editing existing support juts this tad easier.
2. You haven’t run the iProperties script to generate the ready to use property files. Xonx can do this himself, but if there is some error, either a typo you haven’t noticed or a bug in my code triggered by one of your additions, it would be better to catch it before merging into official branch. In this case Xonx should merge it into new branch, then run the script, optionally first fix the alphabetical ordering, and only then merge into unstable.

Both are small things that Xonx can do himself though, so I guess this is indeed ready to be merged, let’s wait and see 😁

Also if Xonx would like a single commit, they can always squash and merge instead of just merging to turn the whole pull request into 1 single commit.

Also now that I think about it, I think the carpets are split between couple different categories, I would need to double check that, and that it most likely my fault, but it will need to be put back together at some point, this is just a note to myself so that I won’t forget :/

Excederus commented:
I tried to follow the already present conventions as close as possible, but didnt even think that alphabetical was a thing to consider haha.

I also ran the script and tested the changes on a running version. I just copied the iProperties files over and then ran the script over there.
However there shouldn't be any typos 🤔 Some stuff can seem like typos (like miniature_azalia - or phoenixbird and rare_phoenix_bird) but that is actually how these are in the mods.

If there is any way to like update the pull request I'm happy to put in the changes tho.

Also these are the Iris errors I get when reloading the shader: https://pastebin.com/Tc0ZUL0W
I am not familiar with Iris errors, but none of them look like they are caused by the changes.

Actually line 877 is missing a \ for the new line, but that is missing in the original (line 832 there) too.

MikiP98 commented:
I think I written the convention in the README in the template folder, though I might have forgot.  

If you tested the changes, pushing the generated properties file is indeed not necessary, Xonx should be able to run it himself with no problems 👍  

If you make another commit to the branch the pull request is from, it should update automatically.  

No idea about the errors. Xonx would need to take a look, but I doubt they are related to properties file.  

Hmm… it is indeed missing, you can add it back in this pull request to not make a new one with just 1 fix.
THX for finding this 😺

Excederus committed:
Fix missing backslash in block.iProperties

Just this for now to test updating the pull request

Excederus committed:
Sort mods alphabetically

Also added #$ comments to new slab variables
Also fixed BLOCK_STAIRS_BOTTOM_INNER_S_E formatting having 1 space too much

Excederus committed:
Forgot to sort the content of $diagonal_walls

Excederus committed:
Support for Luminous Beasts update 1.2.5

Witch Doctor & Woodland Witch Doctor entities and trophy blocks

Excederus committed:
Add support for Connected Glass

Excederus commented:
With my last commit I noticed that diagonalwindows panes have underwater effects applied to them (caustics, low visibility/fog, etc).

This is barely noticable on colored glass, but very noticable on clear glass so i didn't add the clear glass versions to layer.translucent.

I will open an issue tho

Excederus committed:
Add clear diagonal windows to translucent and remove modded diagonal windows due to bug

Excederus committed:
Rename Connected Glass variable to match modid
